tests/components/snmp/test_integer_switch.py

```diff
@@ -0,0 +1,84 @@
+"""SNMP switch tests."""
+
+from unittest.mock import patch
+
+from pysnmp.hlapi import Integer32
+
+from homeassistant.components.switch import DOMAIN as SWITCH_DOMAIN
+from homeassistant.core import HomeAssistant
+from homeassistant.setup import async_setup_component
+
+
+async def test_snmp_integer_switch_off(hass: HomeAssistant) -> None:
+    """Test snmp switch returning int 0 for off."""
+
+    config = {
+        SWITCH_DOMAIN: {
+            "platform": "snmp",
+            "host": "192.168.1.32",
+            # ippower-mib::ippoweroutlet1.0
+            "baseoid": "1.3.6.1.4.1.38107.1.3.1.0",
+            "payload_on": 1,
+            "payload_off": 0,
+        },
+    }
+
+    mock_data = Integer32(0)
+    with patch(
+        "homeassistant.components.snmp.switch.getCmd",
+        return_value=(None, None, None, [[mock_data]]),
+    ):
+        assert await async_setup_component(hass, SWITCH_DOMAIN, config)
+        await hass.async_block_till_done()
+        state = hass.states.get("switch.snmp")
+        assert state.state == "off"
+
+
+async def test_snmp_integer_switch_on(hass: HomeAssistant) -> None:
+    """Test snmp switch returning int 1 for on."""
+
+    config = {
+        SWITCH_DOMAIN: {
+            "platform": "snmp",
+            "host": "192.168.1.32",
+            # ippower-mib::ippoweroutlet1.0
+            "baseoid": "1.3.6.1.4.1.38107.1.3.1.0",
+            "payload_on": 1,
+            "payload_off": 0,
+        },
+    }
+
+    mock_data = Integer32(1)
+    with patch(
+        "homeassistant.components.snmp.switch.getCmd",
+        return_value=(None, None, None, [[mock_data]]),
+    ):
+        assert await async_setup_component(hass, SWITCH_DOMAIN, config)
+        await hass.async_block_till_done()
+        state = hass.states.get("switch.snmp")
+        assert state.state == "on"
+
+
+async def test_snmp_integer_switch_unknown(hass: HomeAssistant) -> None:
+    """Test snmp switch returning int 3 (not a configured payload) for unknown."""
+
+    config = {
+        SWITCH_DOMAIN: {
+            "platform": "snmp",
+            "host": "192.168.1.32",
+            # ippower-mib::ippoweroutlet1.0
+            "baseoid": "1.3.6.1.4.1.38107.1.3.1.0",
+            "payload_on": 1,
+            "payload_off": 0,
+        },
+    }
+
+    mock_data = Integer32(3)
+    with patch(
+        "homeassistant.components.snmp.switch.getCmd",
+        return_value=(None, None, None, [[mock_data]]),
+    ):
+        assert await async_setup_component(hass, SWITCH_DOMAIN, config)
+        await hass.async_block_till_done()
+        state = hass.states.get("switch.snmp")
+        assert state.state == "unknown"
```

Comment:
While I understand the focus of this PR is on testing the existing code.  The existing code should log a warning if it receives a value that is not in range.  Should we add that code fix here or create a new PR and then update the test to check for the warning?

Reply:
Thanks - I'd actually added that in cc252d8ec385338cf4737c8dce07ba777bbea03a in #122944 ; I've cherry-picked that here now.